### PR TITLE
Add StreamJoin + cleanup of StreamFork

### PR DESCRIPTION
**Pull request by piessteady**
Small cleanup of Stream documentation.

### CODE REVIEW

File: source/SpinalHDL/Libraries/stream.rst [hunk 1]
```diff
@@ -425,10 +435,8 @@ This utile take its input stream and duplicate it outputCount times.
 .. code-block:: scala
 
    val inputStream = Stream(Bits(8 bits))
-   val dispatchedStreams = StreamDispatcherSequencial(
-     input = inputStream,
-     outputCount = 3
-   )
+   val (outputstream1, outputstream2) = StreamFork(inputStream,portCount=2)
```
Comment (Dolu1990) [inline source/SpinalHDL/Libraries/stream.rst:438]: This seem broken to me, as StreamFork return a Vec[Stream[T]],
Shouldn't it be either : 

```scala
    val inputStream = Stream(Bool)
    val outputStreams = StreamFork(inputStream,portCount=2)
    // You get outputStreams(0) and outputStreams(1)
```

or 

```scala
    val inputStream = Stream(Bool)
    val (outputstream1, outputstream2)= StreamFork2(inputStream)
```

?

**Comment by Dolu1990** [comment]
Hi ^^
Thanks for the changes :), just one point above about the stream fork seem weird to me, what was the intend ?

File: source/SpinalHDL/Libraries/stream.rst (hunk 1 above)
Comment (piessteady) [reply source/SpinalHDL/Libraries/stream.rst:438]: Yes, wrongly changed from StreamFork2 to StreamFork without checking the return type would be different. Will change it.

**Comment by piessteady** [comment]
Changed and updated the documentation to the comment from the code. Is a better explanation.

**Comment by Dolu1990** [comment]
Thanks :)

**Pull request merged**